src/server/main_service.cc

```diff
@@ -991,27 +991,22 @@ void Service::Eval(CmdArgList args, ConnectionContext* cntx) {
   auto interpreter = ss->BorrowInterpreter();
   absl::Cleanup clean = [ss, interpreter]() { ss->ReturnInterpreter(interpreter); };
 
-  string result;
-  Interpreter::AddResult add_result = interpreter->AddFunction(body, &result);
-  if (add_result == Interpreter::COMPILE_ERR) {
-    return (*cntx)->SendError(result, facade::kScriptErrType);
-  }
+  auto res = server_family_.script_mgr()->Insert(body, interpreter);
+  if (!res)
+    return (*cntx)->SendError(res.error().Format(), facade::kScriptErrType);
 
-  if (add_result == Interpreter::ADD_OK) {
-    if (auto err = server_family_.script_mgr()->Insert(result, body); err)
-      return (*cntx)->SendError(err.Format(), facade::kScriptErrType);
-  }
```

Comment:
All those changes are here, because we need to compile the script _without_ the shebang

Reply:
I do not understand. if it comes from a user why do you need to compile without it?
is it how Redis does it?

Reply:
Yes, its invalid lua code